collect_app/src/main/java/org/odk/collect/android/utilities/SavepointsRepositoryProvider.kt

```diff
@@ -0,0 +1,22 @@
+package org.odk.collect.android.utilities
+
+import android.content.Context
+import org.odk.collect.android.database.savepoints.DatabaseSavepointsRepository
+import org.odk.collect.android.storage.StoragePathProvider
+import org.odk.collect.android.storage.StorageSubdirectory
+import org.odk.collect.forms.savepoints.SavepointsRepository
+
+class SavepointsRepositoryProvider(
+    private val context: Context,
+    private val storagePathProvider: StoragePathProvider
+) {
+
+    @JvmOverloads
+    fun get(projectId: String? = null): SavepointsRepository {
+        val dbPath = storagePathProvider.getOdkDirPath(StorageSubdirectory.METADATA, projectId)
```

Comment:
We could but I'm somehow not sure if this is much better. Handling with resetting is not a big deal so I don't know if it's worth it. Having all databases in one folder seemed to make more sense. Plus placing that database into `cache` dir would mean that we need to create that database from scratch after clearing `cache` (other databases are just cleared).
Do you feel strongly that it would be better?

Reply:
> Do you feel strongly that it would be better?

No not really. It does make sense to me to have all DBs in the same place. I think it'd be good to have a more focused conversation around the file structure as we introduce the entities storage as well, so let's leave this for now.